src/contract.ts

```diff
@@ -2,66 +2,132 @@ import assert from "assert";
+  public async initializeContract(): Promise<AllowedListContract> {
+    const accountRulesList = JSON.parse(
+      fs.readFileSync(
+        path.resolve(__dirname, CONFIG.sidechain.accountRulesList),
+        "utf-8"
+      )
+    );
+    const ruleContractName = await this.ingressContract.methods
+      .RULES_CONTRACT()
+      .call();
+    const accountRulesAddress = await this.ingressContract.methods
+      .getContractAddress(ruleContractName)
+      .call();
 
-        assert(isAddress(accountRulesAddress));
-        this.allowedListContract = new this.web3.eth.Contract(accountRulesList["abi"] as AbiItem[], accountRulesAddress);
+    assert(isAddress(accountRulesAddress));
+    this.allowedListContract = new this.web3.eth.Contract(
+      accountRulesList["abi"] as AbiItem[],
+      accountRulesAddress
+    );
 
-        return this;
+    return this;
+  }
+
+  public async getAccountsList(): Promise<string[] | Error> {
+    try {
+      return await this.allowedListContract.methods.getAccounts().call();
+    } catch (e: any) {
+      const error = e as Error;
+      console.error(error.message);
+      return new Error(
+        `Contract not initialized properly. Details: ${error.message}`
+      );
     }
+  }
 
-    public async getAccountsList(): Promise<string[] | Error> {
-      try {
-        return await this.allowedListContract.methods.getAccounts().call();
-      } catch (e: any) {
-        const error = e as Error;
-        console.error(error.message);
-        return new Error(`Contract not initialized properly. Details: ${error.message}`);
-      }
+  public async getStargateAddress(): Promise<string[] | Error> {
+    try {
+      return await this.bridgeContract.methods.stargateAddress().call();
+    } catch (e: any) {
+      const error = e as Error;
+      console.error(error.message);
+      return new Error(
+        `Contract not initialized properly. Details: ${error.message}`
+      );
     }
+  }
 
-    public async getStargateAddress(): Promise<string[] | Error> {
+  public getAssetIds = async (): Promise<string[] | Error> => {
+    return await this.bridgeContract.methods.getAssetIds().call();
+  };
+  
+  public getTokenRegistryAllowedList = async (): Promise<AssetDetails[] | Error> => {
+    // no filtering for now
+    const assets = await this.getAssetIds();
+    if (assets instanceof Error) return assets;
+  
+    let assetsDetails: AssetDetails[] = [];
+    for (let id of assets) {
       try {
-        return await this.bridgeContract.methods.stargateAddress().call();
-      } catch (e: any) {
-        const error = e as Error;
-        console.error(error.message);
-        return new Error(`Contract not initialized properly. Details: ${error.message}`);
+        const details = await this.bridgeContract.methods.tokenRegistry(id).call();
+  
+        if (details instanceof Error) return details;
+  
+        id = isHexStrict(id) ? id.substring(2) : id; // if 0x is there, then remove it
```

Comment:
sounds good?

Reply:
i think is good! in the future I would opt for what I suggested in slack `remove0x()` which basically encapsulates this same logic inside a handy fn so the code looks cleaner. But this should do the trick 🙌